### PR TITLE
DOC: Better doc for `jnp.hypot`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]
Looks good!

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -3132,9 +3132,38 @@ def heaviside(x1: ArrayLike, x2: ArrayLike, /) -> Array:
...
+    x2: scalar or array. Speicifies the other leg of right angle triangle.
+      ``complex`` dtype are not supported. ``x1`` and ``x2`` must either have
+      same shape or be broadcast compatible.
+
+  Returns:
+    An array containing the hypotenuse for the given given legs ``x1`` and ``x2``
+    of a right angle triangle, promoting to inexact dtype.
+
+  Note:
+    ``jnp.hypot`` is equivalent to computing ``jnp.sqrt(x1 ** 2 + x2 **2)``.
```
Comment (hawkinsp) [inline jax/_src/numpy/ufuncs.py:3154]: But note that's not entirely true! If you compute it that way, it's numerically unstable.

So you should probably say something like "is a more numerically stable way of computing"

File: jax/_src/numpy/ufuncs.py [hunk 2]
```diff
@@ -3132,9 +3132,38 @@ def heaviside(x1: ArrayLike, x2: ArrayLike, /) -> Array:
...
-@implements(np.hypot, module='numpy')
 @jit
 def hypot(x1: ArrayLike, x2: ArrayLike, /) -> Array:
+  r"""
+  Return element-wise hypotenuse for the given legs of a right angle triangle.
+
+  JAX implementation of :obj:`numpy.hypot`.
+
+  Args:
+    x1: scalar or array. Speicifies one of the legs of right angle triangle.
```
Comment (hawkinsp) [inline jax/_src/numpy/ufuncs.py:3143]: Typo: Specifies

Same below.

Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Thanks! Fixed the typo.

File: jax/_src/numpy/ufuncs.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Thanks! Updated the note.

**Review by hawkinsp — APPROVED** [review]

**Pull request merged**